northstar/src/runtime/process/mod.rs

```diff
@@ -278,27 +299,45 @@ fn waitpid(
     pid: Pid,
     tx: EventTx,
     stop: CancellationToken,
+    mut channel: pipe::AsyncChannelRead,
```

Comment:
> after introducing this channel and removing the wait for `sigchld`, I'm not sure if the name `waitpid` is still valid for this function.

I kept the name `waitpid` intentionally because every system programmer knows what a `waitpid` does.

> I would actually suggest to inline this function. It would also nice to have some more verbosity on the variable names handled here. Using `pid`, `channel` and `exit_status` is confusing after so many `forks`. Maybe we could have something like:
> 
> ```rust
>    let process_exit_status = child_channel.rec::<ExitStatus>();
>    let init_exit_status = waitpid(init_pid);
> 
>     // here goes some explanation to why this is the way it is
>     let exit_status_fut = process_exit_status.or_else(|error| {
>         warn!("failed to ... {}" error);
>         init_exit_status
>     });
> 
>    let notify_on_exit_task = task::spwan(async move {
>        let exit_status = exit_status_fut.await.expext("no exit status");
>        stop.cancel();
>        tx.send(exit_status).await.ok();
>       // ...
>    });
> ```

Ok. I'll get back on this and try to sort things a little bit. Probably I spent too much time in this code and some things are obvious for me which aren't...